backend/app/analysis/tests.py

```diff
@@ -32,19 +36,117 @@ def setUp(self):
+            if image.path.endswith('.jpg') and image.is_file():  # Ensures file is an image
+                # Creates Photo object
+                photo = Photo(number=photo_number, map_square=self.map_square)
+                photo.front_local_path = image.path
+                photo.save()
 
-        self.photo_1 = Photo(number=2, map_square=self.map_square)
-        self.photo_1.save()
+                # Splits path string to make Photo name
+                photo_name = image.path.split(str(settings.TEST_PHOTOS_DIR)+'/')[1].split('.jpg')[0]
+                self.photo_dict[photo_name] = photo
 
     def test_photographer_caption_length(self):
-        self.photo_0.photographer_caption = '123456'
-        result = photographer_caption_length.analyze(self.photo_0)
-        self.assertEqual(6, result)
+        """
+        Test Photographer Caption Length analysis (photographer_caption_length.py)
+        """
+        expected_values = {'100x100_500px-white_500px-black': 6}
+        for image in expected_values:
+            if image in self.photo_dict.keys():
+                self.photo_dict[image].photographer_caption = '123456'
+                result = photographer_caption_length.analyze(self.photo_dict[image])
+                print(f'Caption Length performed on {image}. Result: {result}')
+                self.assertEqual(expected_values[image], result)
 
     def test_whitespace_percentage(self):
-        result = whitespace_percentage.analyze(self.photo_0)
-        self.assertEqual(50, result)
+        """
+        Test Whitespace Percentage analysis (whitespace_percentage.py)
+        """
+        expected_values = {'100x100_500px-white_500px-black': 50}
+        for image in expected_values:
+            if image in self.photo_dict.keys():
+                result = whitespace_percentage.analyze(self.photo_dict[image])
+                print(f'Whitespace Percentage performed on {image}. Result: {result}')
+                self.assertEqual(expected_values[image], result)
+
+    def test_stdev(self):
+        """
+        Test Standard Deviation analysis (stdev.py)
+        """
+        expected_values = {'100x76-BlackRectangle': 0, '100x76-GreyRectangle': 0,
+                           '100x76-WhiteRectangle': 0, '100x100-BlackSquare': 0,
+                           '100x100-GreySquare': 0, '100x100-WhiteSquare': 0,
+                           '100x76-CheckeredRectangle_1': 126, '100x76-CheckeredRectangle_2':
+                               126, '100x76-HalfRectangle_1': 1, '100x76-HalfRectangle_2': 126,
+                           '100x76-HalfRectangle_3': 1, '100x76-HalfRectangle_4': 126,
+                           '100x100-CheckeredSquare_1': 125, '100x100-CheckeredSquare_2': 125,
+                           '100x100-HalfSquare_1': 0, '100x100-HalfSquare_2': 125,
+                           '100x100_500px-white_500px-black': 0,
+                           '100x100-HalfSquare_4': 125}
+
+        for image in expected_values:
```

Comment:
here the for loop makes perfect sense, but the `if` has the same problem: if the image isn't there, we want this test to crash because something has gone wrong.